src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -182,34 +183,71 @@ class PosHashTable {
   std::optional<std::vector<std::pair<HashedType, Offset>>> _values{std::nullopt};
 };
 
+// The bloom filter (with k=1) is used during the materialization and build phases. It contains `true` for each
+// `hash_function(value) % BLOOM_FILTER_SIZE`. Much of this could be improved, for example by (1) dynamically checking
+// whether bloom filters make sense, (2) choosing an appropriate filter size, and (3) working with multiple hash
+// functions.
+static constexpr auto BLOOM_FILTER_BITS = 20;
+static constexpr auto BLOOM_FILTER_SIZE = 1 << BLOOM_FILTER_BITS;
```

Comment:
Any reason not to use table size information here? For small tables, the bloom filter unnecessarily leaves the L1 cache, doesn't it?

Reply:
Because it would not be constexpr anymore and we would have to pass information around. If we start doing something with the table size here, it would have to be somewhat reasonable. A heuristic that looks smart but is half-assed seems worse than admittedly using a constent size.

I realize that I come off as quite negative in these responses. You raise very valid issues - it's just that I decided to personally pause the iter/s chase once the TPC-H paper is done. I'll be happy to help if someone wants to pick up where I stopped. For now, it's either getting this PR to a mergeable state (with known shortcomings) or forgoing the substantial the benefits altogether.

Reply:
I totally understand that. But I wonder where we can somehow collect all the brought up points (let it be NUMA, the hash method, the bloom filter sizes) to allow somebody else to improve the join.

Reply:
Added a list.